packages/xsnap/src/avaAssertXS.js

```diff
@@ -0,0 +1,315 @@
+/* eslint-disable no-await-in-loop */
+// @ts-check
+
+/** global print */
+
+const { freeze, keys } = Object;
+
+// ack Paul Roub Aug 2014
+// https://stackoverflow.com/a/25456134/7963
+/** @type {(x: unknown, y: unknown) => boolean } */
+function deepEqual(x, y) {
+  if (x === y) {
```

Comment:
@erights you noted this use of `===` in `deepEqual` was surprising and asked if it's consistent with ava.

No, it's not. [ava's deepEqual](https://github.com/avajs/ava/blob/main/lib/assert.js#L370-L388) uses [concordance](https://github.com/concordancejs/concordance#readme), which is different in at least one way: `NaN` equals `NaN`.